### PR TITLE
New tile structure + design

### PR DESCRIPTION
**Pull request by peuter**

### CODE REVIEW

File: source/class/cv/ConfigCache.js [hunk 1]
```diff
@@ -157,6 +157,9 @@ qx.Class.define('cv.ConfigCache', {
             });
           }, this);
         }
+        // hide body to prevent flickering
+        body.style.visibility = 'hidden';
+        console.log('### visibility', body.style.visibility);
```
Comment (ChristianMayer) [inline source/class/cv/ConfigCache.js:162]: On purpose - or left over?

File: source/class/cv/ui/structure/tile/Controller.js [hunk 2]
```diff
@@ -0,0 +1,492 @@
+/**
```
Comment (ChristianMayer) [inline source/class/cv/ui/structure/tile/Controller.js:1]: Copyright info missing

File: source/class/cv/ui/structure/tile/Controller.js [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+    onComponentCreated(element) {
+      const name = element.tagName.toLowerCase();
+      if (Object.prototype.hasOwnProperty.call(this, name)) {
+        const QxClass = this.__MAP[name];
+        if (!Object.prototype.hasOwnProperty.call(this, name)) {
+          this.__I[name] = [];
+        }
+        this.__I[name].push(new QxClass(element));
+      } else {
+        qx.log.Logger.error('no QxClass registered for custom element ' + name);
```
Comment (ChristianMayer) [inline source/class/cv/ui/structure/tile/Controller.js:62]: can't this be written as `this.error()` instead?

File: source/class/cv/ui/structure/tile/Controller.js [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+          }
+          document.body.classList.remove('loading-structure');
+        }, this);
+
+        ajaxRequest.send();
+      }
+    },
+
+    /**
+     * Reisters customElements for all templates in the given xml that are direct children of a <templates structure="tile"> element
```
Comment (ChristianMayer) [inline source/class/cv/ui/structure/tile/Controller.js:236]: Registers

File: source/class/cv/ui/structure/tile/components/Color.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
+/**
+ * Shows a html color input
+ * @author Tobias Bräutigam
+ * @since 2022
+ */
+qx.Class.define('cv.ui.structure.tile.components.Color', {
```
Comment (ChristianMayer) [inline source/class/cv/ui/structure/tile/components/Color.js:6]: Due to the (Professional) ColorChooser the CV has a quite thorough color handling class. Probably it should be used here - especially as the visu color input will most likely be used for lighting applications where the basic web assumption "everything is sRGB" doesn't hold anymore

File: source/class/cv/ui/structure/tile/Controller.js (hunk 3 above)
Comment (peuter) [reply source/class/cv/ui/structure/tile/Controller.js:62]: No, `this.error` is only available in class instances, this is a static function where you have to use `qx.log.Logger.error(this, <error-message>` but I forgot the `this`, so I will fix that.

File: source/class/cv/ui/structure/tile/components/Color.js (hunk 5 above)
Comment (peuter) [reply source/class/cv/ui/structure/tile/components/Color.js:6]: Please ignore the current state of the color component, its not even a proof-of-concept more a proof-that-the-html-color-input-is-no-solution-for-this. I haven't decided yet which way I want ot go with the color-component here. but I do not want to have the complexity of the existing ColorChooser here. I want something very simple, which will fit my needs and probably the needs of must of the users. Might be an option to implement a way to integrate the professional colorchooser for users the need more features, but thats something for "later".

File: source/test/karma/ui/structure/pure/ColorChooser-spec.js [hunk 6]
```diff
@@ -36,7 +36,7 @@ describe('testing a colorchooser', function() {
 
   // test bug reported here:
   // https://knx-user-forum.de/forum/supportforen/cometvisu/963486-problem-milight-colorchooser-mit-oh-als-backend
-  it('should test the colorchooser with incoming data in RGB mode', function() {
```
Comment (ChristianMayer) [inline source/test/karma/ui/structure/pure/ColorChooser-spec.js:39]: Name / title doesn't fit to content anymore. The variant is `rgb`

**Comment by ChristianMayer** [comment]
Testing it I get  those errors
```
GET https://timberwolf76.local/proxy/visugit/obiwan/designs/tile/basic.css?1661445025076 net::ERR_ABORTED 404
005999 cv.util.ScriptLoader: error loading designs/tile/basic.css?1661445025076
```

**Comment by peuter** [comment]
> Testing it I get those errors
> ...

I know, thats related to the fact that I use SCSS here, I have to find a solution for that problem. But the actual correct css file is loaded (or do you experience something else?). So that is just about avoiding an unneeded HTTP-call that ends in an 404 error and not about fixing somehting that is really broke. And therefore it is not on top of my priority list.

**Comment by ChristianMayer** [comment]
Opening `?config=demo-tile#start` gives the red error popup:
```
Das Backend hat eine ungültige Antwort für eine Lese-Anfrage gesendet: Fehlender "i" Wert.
Bitte versuchen Sie das Problem im Backend zu lösen.
```

**Comment by peuter** [comment]
I need more details about:
* the console errors
* the URL the read request was sent too (is it the right one?)
* how does the read response look like, or is it empty?

**Comment by ChristianMayer** [comment]
Playing with the demo config and changing pages I get this error message on the console:
```
jquery.flot.min.js?1661719180069:7 Uncaught Error: Time mode requires the flot.time plugin.
    at setupTickGeneration (jquery.flot.min.js?1661719180069:7:28077)
    at Object.<anonymous> (jquery.flot.min.js?1661719180069:7:26263)
    at Function.each (jquery.js:365:19)
    at Plot.setupGrid (jquery.flot.min.js?1661719180069:7:26227)
    at wrapper.<anonymous> (AbstractDiagram.js:807:18)
    at Function.lookupTsCache (AbstractDiagram.js:319:9)
    at wrapper.<anonymous> (AbstractDiagram.js:766:44)
    at Array.forEach (<anonymous>)
    at wrapper.loadDiagramData (AbstractDiagram.js:761:28)
    at wrapper.<anonymous> (Diagram.js:99:18)
    at wrapper.__callSubscribers__P_104_1 (Bus.js:421:24)
    at wrapper.dispatch (Bus.js:348:16)
    at wrapper.dispatchByName (Bus.js:368:22)
    at Function.dispatchByName (Bus.js:112:48)
    at wrapper.seekTo (PageHandler.js:81:28)
    at wrapper.scrollToPage (Controller.js:363:24)
    ```
    After that a change to the root page doesn't work anymore.
    
    A crosscheck with the current develop branch didn't show this error message

**Comment by peuter** [comment]
The flot error should be fixed, the random clicking one is not reproducible for me. Lets hope that is just has been a follow up error to the flot problem and is then fixed too.

**Comment by ChristianMayer** [comment]
The flot error doesn't come anymore. Also not the getCss one. 
But opening the demo config on the start page gives:
```
AbstractDiagram.js:337 Uncaught TypeError: Cannot read properties of undefined (reading '0')
    at Function._onSuccess (AbstractDiagram.js:337:71)
    at Function.<anonymous> (AbstractDiagram.js:310:18)
    at Direct.js:136:37
    at Function.then (Utils.js:159:39)
    at Direct.js:135:26
    at Array.forEach (<anonymous>)
    at wrapper.dispatchEvent (Direct.js:120:19)
    at wrapper.wrappedFunction [as dispatchEvent] (Interface.js:537:31)
    at wrapper.dispatchEvent (Manager.js:975:50)
    at Registration.js:346:42
    at Function.then (Utils.js:159:39)
    at Object.fireEvent (Registration.js:345:22)
    at wrapper.fireEvent (MEvent.js:182:36)
    at wrapper._fireStatefulEvent (AbstractRequest.js:928:12)
    at wrapper.__onReadyStateDone__P_249_12 (AbstractRequest.js:849:16)
    at wrapper._onReadyStateChange (AbstractRequest.js:821:14)
```
When I open it from a sub page (e.g. `?config=demo#id_40_`) that error doesn't come. Also going back to the start page it doesn't come.
This is reproducible.

File: doc/manual/de/config/structure-tile/index.rst [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+                <cv-address transform="OH:number" mode="read" backend="si">Test_Value</cv-address>
+                <cv-round-progress class="value"/>
+            </cv-value>
+        </cv-row>
+        <cv-row colspan="3" row="last">
+            <label class="secondary">Circle Progress</label>
+        </cv-row>
+    </cv-tile>
+
+Dieser Kachel-Konfiguration muss nun in ein Template übertragen werden. Dazu muss man sich zunächst eine Template-Datei
```
Comment (ChristianMayer) [inline doc/manual/de/config/structure-tile/index.rst:250]: Diese

File: doc/manual/de/config/structure-tile/index.rst [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+hinzu.
+
+.. code-block:: xml
+
+    <cv-meta>
+        <cv-loader type="templates" src="resource/config/media/my-templates.xml" />
+    </cv-meta>
+
+
+Diese Datei sollte dann folgenden Inhalt enthalten.
```
Comment (ChristianMayer) [inline doc/manual/de/config/structure-tile/index.rst:262]: `.` -> `:`

File: doc/manual/de/config/structure-tile/index.rst [hunk 9]
```diff
@@ -0,0 +1,335 @@
...
+    </templates>
+
+Man sieht innerhalb der Template-Definition mit der id "meter" den leicht modifizierten Inhalt der Kachel.
+Modifiziert werden müssen lediglich Inhalte die man später konfigurierbar haben möchte. In der Regel
+gehören dazu Adressen, Label und ggf. einige Attribute wie Mappings, Stylings und Formatierungen.
+
+Wenn man nun dieses Template in der Konfigurationsdatei benutzen möchte fügt man dort ein ``<cv-meter>`` Element hinzu.
+Der Name ergibt sich aus der Template ID mit dem Präfix ``cv-``.
+
+Das Prinzip der <slot>-Elemente ist relativ einfach, sie dienen als Platzhalter für Elemente die diesen Slot benutzen.
```
Comment (ChristianMayer) [inline doc/manual/de/config/structure-tile/index.rst:289]: `<slot>` -> ` ``<slot>`` `

File: source/class/cv/io/Net.js [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+  /*
+  ***********************************************
+    STATICS
+  ***********************************************
+  */
+  statics: {
+    /**
+     * Very basic window.fetch replacement that does not support the `init` parameter and therefore only supports
+     * GET requests. Uses `qx.io.request.Xhr` internally which supports the XHR calls being recorded / replayed
+     * by CometVisus own recording feature. window.fetch calls are nor supported by the recording feature and get lost.
```
Comment (ChristianMayer) [inline source/class/cv/io/Net.js:35]: nor -> not

File: source/class/cv/io/System.js [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
+ */
+
+/**
+ * A Backend client that handles write so internal system calls, e.g. to restart a backend connection, reload the browser, etc.
```
Comment (ChristianMayer) [inline source/class/cv/io/System.js:21]: Please rephrase sentence

**Review by ChristianMayer — CHANGES_REQUESTED** [review]
What is the license of `remixicon`? Should it be added to its directoy?

File: doc/manual/de/config/structure-tile/elements/backend.rst [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+
+    <cv-button size="small" style="position: absolute; top: 0; right: 0; color: red">
+      <cv-address mode="write" backend="system">https://somewhere/delete?this</cv-address>
+      <cv-icon class="ri-delete-bin-line"/>
+    </cv-button>
+
+Oft kommt es beim Aufrufen von URLs von externen Webseiten zu einen CORS Fehler, da das Aufrufen von externen Seiten
+ein Sicherheitsrisiko darstellt. Dies kann umgangen werden, indem die interne Proxy-Funktion dder CometVisu benutzt wird.
+Dazu müssen die aufzurufenden URLs mit ":proxy" erweitert werden. Für das o.g. Beispiel würde aus
+``https://somewhere/delete?this`` -> ``https:proxy://somewhere/delete?this``.
```
[the remark this answers is not in the document]
Comment (peuter) [reply doc/manual/de/config/structure-tile/elements/backend.rst:170]: Das funktioniert aber anders herum, die keywords am Anfang sagen dem System-Backend was zu tun ist (also http/https sendet HTTP-Requests). Was danach kommt konfiguriert diesen Request.
Für anderen Protokolle muss sowieso ein anderer Handler im System-Backend implementiert werden und der müsste dann ggf. auch die Proxy-Geschichte implementieren.

**Comment by peuter** [comment]
The AbstractDiagram Error seems to be related to the received data. As I don't have any RRDs here I only get an 404 Error. If I create an empty file I get no error in the UI  (but the response contains `[[ERROR000,["mmaping","file","'/var/www/rrd/eib_traffic.rrd'","Invalid","argument"]]]`).
So I don't know how to reproduce this problem.

**Comment by ChristianMayer** [comment]
Using mobile device simulation mode an making the view (very) small triggers lots of error messages:
image

**Comment by ChristianMayer** [comment]
Except the last point from 21.09. this PR is ready to be merged

**Pull request merged**